crates/core/src/syn/parser/stmt/alter.rs

```diff
@@ -79,25 +75,132 @@ impl Parser<'_> {
 					self.pop_peek();
 					res.permissions = Some(ctx.run(|ctx| self.parse_permission(ctx, false)).await?);
 				}
 				t!("CHANGEFEED") => {
 					self.pop_peek();
-					if self.eat(t!("NONE")) {
-						res.changefeed = Some(None);
+					res.changefeed = Some(if drop {
+						None
 					} else {
-						res.changefeed = Some(Some(self.parse_changefeed()?));
+						Some(self.parse_changefeed()?)
+					})
+				}
+				_ if drop => {
+					unexpected!(self, peek, "`COMMENT` or `CHANGEFEED`")
+				}
+				_ => break,
+			}
+		}
+
+		Ok(res)
+	}
+
+	pub(crate) async fn parse_alter_field(
+		&mut self,
+		ctx: &mut Stk,
+	) -> ParseResult<AlterFieldStatement> {
+		let if_exists = if self.eat(t!("IF")) {
+			expected!(self, t!("EXISTS"));
+			true
+		} else {
+			false
+		};
+		let name = self.parse_local_idiom(ctx).await?;
+		expected!(self, t!("ON"));
+		self.eat(t!("TABLE"));
+		let what = self.next_token_value()?;
+		let mut res = AlterFieldStatement {
+			name,
+			what,
+			if_exists,
+			..Default::default()
+		};
+
+		loop {
+			let drop = self.eat(t!("DROP"));
+			let peek = self.peek();
+			match peek.kind {
+				t!("FLEXIBLE") => {
+					self.pop_peek();
+					res.flex = Some(!drop)
+				}
+				t!("TYPE") => {
+					self.pop_peek();
+					res.kind = Some(if drop {
+						None
+					} else {
+						Some(ctx.run(|ctx| self.parse_inner_kind(ctx)).await?)
+					});
+				}
+				t!("READONLY") => {
+					self.pop_peek();
+					res.flex = Some(!drop)
+				}
+				t!("VALUE") => {
+					self.pop_peek();
+					res.value = Some(if drop {
+						None
+					} else {
+						Some(ctx.run(|ctx| self.parse_value_field(ctx)).await?)
+					});
+				}
+				t!("ASSERT") => {
+					self.pop_peek();
+					res.assert = Some(if drop {
+						None
+					} else {
+						Some(ctx.run(|ctx| self.parse_value_field(ctx)).await?)
+					});
+				}
+				t!("DEFAULT") => {
+					self.pop_peek();
+
+					if drop {
+						res.default = Some(None);
+					} else {
+						res.default_always = Some(self.eat(t!("ALWAYS")));
+						res.default = Some(Some(ctx.run(|ctx| self.parse_value_field(ctx)).await?));
+					}
+				}
+				t!("PERMISSIONS") if !drop => {
+					self.pop_peek();
+					res.permissions = Some(ctx.run(|ctx| self.parse_permission(ctx, false)).await?);
+				}
+				t!("COMMENT") => {
+					self.pop_peek();
+					res.comment = Some(if drop {
+						None
+					} else {
+						Some(self.next_token_value()?)
+					})
+				}
+				t!("REFERENCE") => {
+					if !self.settings.references_enabled {
+						bail!(
+							"Experimental capability `record_references` is not enabled",
+							@self.last_span() => "Use of `REFERENCE` keyword is still experimental"
+						)
 					}
+
+					self.pop_peek();
+					res.reference = Some(if drop {
+						None
+					} else {
+						Some(self.parse_reference(ctx).await?)
+					});
+				}
+				_ if drop => {
+					unexpected!(self, peek, "`FLEXIBLE`, `TYPE`, `READONLY`, `VALUE`, `ASSERT`, `DEFAULT`, `COMMENT`, or `REFERENCE`")
```

Comment:
Same here, please just parse by matching a keyword and then matching what is allowed after.